File: common/src/main/java/com/ultreon/mods/advanceddebug/api/common/AbstractSize.java
package com.ultreon.mods.advanceddebug.api.common;

import java.math.BigDecimal;
import java.math.BigInteger;

public abstract class AbstractSize {
    public abstract Number getWidth();

    public abstract Number getHeight();

    public int getIntWidth() {
        return (int) this.getWidth();
    }

    public int getIntHeight() {
        return (int) this.getHeight();
    }

    public long getLongWidth() {
        return (long) this.getWidth();
    }

    public long getLongHeight() {
        return (long) this.getHeight();
    }

    public float getFloatWidth() {
        return (float) this.getWidth();
    }

    public float getFloatHeight() {
        return (float) this.getHeight();
    }

    public double getDoubleWidth() {
        return (double) this.getWidth();
    }

    public double getDoubleHeight() {
        return (double) this.getHeight();
    }

    public BigInteger getBigIntegerWidth() {
        return new BigInteger(this.getWidth().toString());
    }

    public BigInteger getBigIntegerHeight() {
        return new BigInteger(this.getHeight().toString());
    }

    public BigDecimal getBigDecimalWidth() {
        return new BigDecimal(this.getWidth().toString());
    }

    public BigDecimal getBigDecimalHeight() {
        return new BigDecimal(this.getHeight().toString());
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/common/AbstractSize.kt
package com.ultreon.mods.advanceddebug.api.common

import java.math.BigDecimal
import java.math.BigInteger

abstract class AbstractSize {
    abstract val width: Number
    abstract val height: Number
    val intWidth: Int
        get() = width as Int
    val intHeight: Int
        get() = height as Int
    val longWidth: Long
        get() = width as Long
    val longHeight: Long
        get() = height as Long
    val floatWidth: Float
        get() = width as Float
    val floatHeight: Float
        get() = height as Float
    val doubleWidth: Double
        get() = width as Double
    val doubleHeight: Double
        get() = height as Double
    val bigIntegerWidth: BigInteger
        get() = BigInteger(width.toString())
    val bigIntegerHeight: BigInteger
        get() = BigInteger(height.toString())
    val bigDecimalWidth: BigDecimal
        get() = BigDecimal(width.toString())
    val bigDecimalHeight: BigDecimal
        get() = BigDecimal(height.toString())
}